# Marriott Hotel Category Change 2020 Analysis

In [1]:
# import pandas
import pandas as pd

In [2]:
# read data from csv
df = pd.read_csv('marriott-category-changes-2020.csv')
df.head()

Hotel                  Brand     Destination  \
0     Aberdeen Marriott Hotel               Marriott  United Kingdom   
1             AC Hotel Aitana  AC Hotels by Marriott           Spain   
2  AC Hotel Alcala de Henares  AC Hotels by Marriott           Spain   
3            AC Hotel Almeria  AC Hotels by Marriott           Spain   
4            AC Hotel Aravaca  AC Hotels by Marriott           Spain   

   Current Category  Current Standard Price  New Category  New Standard Price  
0                 3                   17500             2               12500  
1                 3                   17500             4               25000  
2                 1                    7500             2               12500  
3                 1                    7500             2               12500  
4                 1                    7500             2               12500

## Question 1
Discribe the data types for each feature/column, e.g., xxx feature's data type is String, yyy feature's data type is float, etc.

In [3]:
# prints out data types for each columns.
for colname, coltype in df.dtypes.iteritems():
    print(colname,'data type is',coltype)

Hotel data type is object
Brand data type is object
Destination data type is object
Current Category data type is int64
Current Standard Price data type is int64
New Category data type is int64
New Standard Price data type is int64


My answer to the question 1:
* Hotel data type is object
* Brand data type is object
* Destination data type is object
* Current Category data type is int64
* Current Standard Price data type is int64
* New Category data type is int64
* New Standard Price data type is int64

## Question 2
- How many hotels are in this dataset?
- The hotels are from how many unique brands?
- Which destination/country has the most hotels listed in this dataset? List the total number of hotels in that country
- How many brands in China have hotel category changes?

In [4]:
# getting data shape, listing total number of rows and columns.
df.shape

(2185, 7)

In [5]:
# list all unique values in brand columns
print(df['Brand'].value_counts())

# prints out total number of brands.
print('The hotels are from',df['Brand'].nunique(),'unique brands.')

Courtyard                      366
Fairfield Inn                  338
Residence Inn                  273
Sheraton                       154
Marriott                       148
TownePlace Suites              131
SpringHill Suites              130
Four Points                     93
Autograph Collection            69
Westin                          67
AC Hotels by Marriott           54
Aloft                           52
Renaissance                     45
Le Meridien                     34
Luxury Collection               33
Protea Hotels                   30
Ritz-Carlton                    29
JW Marriott                     23
Element                         22
Delta                           22
Moxy Hotels                     21
W Hotels                        16
Tribute Portfolio               13
St. Regis                        9
VISTA                            3
Marriott Vacation Club Intl      3
Edition                          3
Design Hotels                    2
Gaylord Hotels      

In [6]:
# list the total numbers of hotel bt their destination. showing top 5 destinations.
df2 = df.groupby('Destination')['Hotel'].nunique().sort_values(ascending=False).reset_index(name='count')
df2

Destination  count
0           USA   1545
1        Canada     76
2         China     68
3         Spain     34
4        Mexico     26
..          ...    ...
93      Denmark      1
94  New Zealand      1
95      Curacao      1
96       Israel      1
97       Norway      1

[98 rows x 2 columns]

In [7]:
# create a data subset which only conain China as destination.then show the total brands in the dataset.
cdf = df2.loc[df2['Destination'] == "China"]
cdf.head()

Destination  count
2       China     68

In [8]:
# this part is only for testing the results
# count the totla number of hotels which meets the condition: current category not equal to new category.
counter = 0
for i,r in df.iterrows():
       if (r['Current Category'] != r['New Category']) & (r['Destination'] == 'China'):
        counter+=1
# There are total 68 hotels have category changed in China.
counter

68

My answer to the question 2:
* There are 2185 hotels in this dataset.
* The hotels are from 30 unique brands.
* USA has the most number of hotels in this dataset which are 1545 hotels.
* There are 68 hotels in China have category changed.

## Question 3
- What's the percentage of hotels worldwide with category upgrade in 2020?

In [29]:
# first count how many hotels have category increased worldwide.
# create a new column shows total category increased.
df['total_upcatchanged']=df['New Category'] - df['Current Category']

# sliced the data which contains total category at least 1, and then count the total number of hotels.
updf = df.loc[df['total_upcatchanged'] >= 1]

# divided by its total number of hotels 
uppercen = round((updf.shape[0]/df.shape[0])*100,2)
print('There are %{} of hotels worldwide with category upgrade in 2020.' .format(uppercen))

There are %77.16 of hotels worldwide with category upgrade in 2020.


My answer to the question 3:
* There are %77.16 of hotels worldwide with category upgrade in 2020.

## Question 4
- List hotels with category changes greater than 1 if any, such as changing from category 3 to 5 or from category 7 to 4
- List all JW Marriott hotels in China that have a category upgrade

In [30]:
# create a new column shows total category change.
df['total_changed'] = abs(df['New Category'] - df['Current Category'])
print(df['total_changed'].value_counts())

# List hotels with category changes greater than 1 if any and subset it into another dataframe.
morechangedf = df.loc[df['total_changed']>1]
morechangedf.head()

1    2184
2       1
Name: total_changed, dtype: int64


Hotel        Brand Destination  \
913  Four Points by Sheraton Bali, Ungasan  Four Points   Indonesia   

     Current Category  Current Standard Price  New Category  \
913                 4                   25000             2   

     New Standard Price  total_upcatchanged  total_catchanged  total_changed  
913               12500                  -2                 2              2

In [23]:
a = df.loc[(df['Brand']=='JW Marriott') & (df['Destination']=='China') & (df['total_upcatchanged'] >= 1)]
a

Hotel        Brand Destination  \
1074          JW Marriott Hotel Chengdu  JW Marriott       China   
1078         JW Marriott Hotel Shenzhen  JW Marriott       China   
1079  JW Marriott Hotel Shenzhen Bao'an  JW Marriott       China   
1083        JW Marriott Hotel Zhengzhou  JW Marriott       China   

      Current Category  Current Standard Price  New Category  \
1074                 3                   17500             4   
1078                 3                   17500             4   
1079                 3                   17500             4   
1083                 2                   12500             3   

      New Standard Price  total_upcatchanged  total_catchanged  
1074               25000                   1                 1  
1078               25000                   1                 1  
1079               25000                   1                 1  
1083               17500                   1                 1

My answer to the question 4:
* There is only 1 hotel with category changes greater than 1 which is hotel Four Points by Sheraton Bali, Ungasan. 
* All JW Marriott hotels in China that have a category upgrade is listed above.

## Question 5
Assume you are in Feburary 2020 and the category changes will take effect on March 4, 2020. You are planning your trip to Florence, Italy and Hong Kong, China in April. You only stay in category 8 hotel (existing category 8 or future category 8) and want to optimize your point spending. Based on the data, which hotel you should book? when should you book your hotels for Florence and Hong Kong? Why?

In [36]:
# get all the hotels which located in Italy and have or will have category level 8.
b = df.loc[((df['Destination']=='Italy') & (df['Current Category']==8)) | ((df['Destination']=='Italy') & (df['New Category']==8))]
b

Hotel              Brand  \
145                  Cervo Hotel, Costa Smeralda Resort           Sheraton   
519   Cristallo, a Luxury Collection Resort & Spa, C...  Luxury Collection   
1968                     The Westin Excelsior, Florence             Westin   

     Destination  Current Category  Current Standard Price  New Category  \
145        Italy                 7                   60000             8   
519        Italy                 8                   85000             7   
1968       Italy                 7                   60000             8   

      New Standard Price  total_upcatchanged  total_catchanged  total_changed  
145                85000                   1                 1              1  
519                60000                  -1                 1              1  
1968               85000                   1                 1              1

In [39]:
# get all the hotels which located in Hong Kong and have or will have category level 8.
c = df.loc[((df['Destination']=='Hong Kong') & (df['Current Category']==8)) | ((df['Destination']=='Hong Kong') & (df['New Category']==8))]
c

Empty DataFrame
Columns: [Hotel, Brand, Destination, Current Category, Current Standard Price, New Category, New Standard Price, total_upcatchanged, total_catchanged, total_changed]
Index: []

My answer to the question 5:
* I would like to book Cervo Hotel, Costa Smeralda Resort and The Westin Excelsior, Florence before March 4, 2020. This is because these two hotels are going to upgrade to category 8 on March. However,the price for these two hotels are still cheap right now, which are 60000, and it will increase to 85000 once it upgrade to category 8. So I would like to book these two hotels before March 4. On the other side, I would not book any hotels in Hong Kong since there are no any category 8 hotel in this place. 